### PR TITLE
Nests `CombineReference` inside of `WriteCombineReference` Transform

### PR DESCRIPTION
**Pull request by norlandrhagen**
PR for issue: https://github.com/pangeo-forge/pangeo-forge-recipes/issues/634

cc @cisaacstern

### CODE REVIEW

File: pangeo_forge_recipes/transforms.py [hunk 1]
```diff
@@ -448,6 +448,18 @@ def expand(self, references: beam.PCollection) -> beam.PCollection:
 @dataclass
 class WriteCombinedReference(beam.PTransform, ZarrWriterMixin):
     """Store a singleton PCollection consisting of a ``kerchunk.combine.MultiZarrToZarr`` object.
+    :param concat_dims: Dimensions along which to concatenate inputs.
```
Comment (cisaacstern) [inline pangeo_forge_recipes/transforms.py:451]: Looks like this is duplicated in the docstring (see bottom of param list).

File: pangeo_forge_recipes/transforms.py [hunk 2]
```diff
@@ -457,14 +469,24 @@ class WriteCombinedReference(beam.PTransform, ZarrWriterMixin):
     .parquet as a storage format.
     """
 
-    store_name: str
+    concat_dims: List[str] = field(default_factory=list)
+    identical_dims: List[str] = field(default_factory=list)
+    mzz_kwargs: dict = field(default_factory=dict)
+    precombine_inputs: bool = False
+    store_name: Optional[str] = None
```
Comment (cisaacstern) [inline pangeo_forge_recipes/transforms.py:476]: `store_name` is required so let's not make that Optional if at all possible.

As for `concat_dims` & `identical_dims`, I think those are both required for `CombineReferences` so I think we should not be using a default factory. If either of these kws are not passed, we _do want_

```python
TypeError: __init__() missing 1 required positional argument:
```
to be raised, which the default factory will prevent.

File: pangeo_forge_recipes/transforms.py [hunk 3]
```diff
@@ -429,8 +429,8 @@ class CombineReferences(beam.PTransform):
       dimensionality will match that of the accumulator.
     """
 
-    concat_dims: List[str]
-    identical_dims: List[str]
+    concat_dims: List[str] = field(default_factory=list)
+    identical_dims: List[str] = field(default_factory=list)
```
Comment (cisaacstern) [inline pangeo_forge_recipes/transforms.py:433]: Let's not use default factories here if at all possible.

IIUC these are required, so if they are not passed, we _do want_ a TypeError to be raised (which the default factory will prevent).

See related comment below.

File: tests/test_end_to_end.py [hunk 4]
```diff
@@ -148,11 +146,9 @@ def test_reference_grib(
             p
             | beam.Create(pattern.items())
             | OpenWithKerchunk(file_type=pattern.file_type)
-            | CombineReferences(
+            | WriteCombinedReference(
                 concat_dims=[pattern.concat_dims[0]],
                 identical_dims=["latitude", "longitude"],
-            )
-            | WriteCombinedReference(
```
Comment (cisaacstern) [inline tests/test_end_to_end.py:155]: This is beautiful diff 💐 😄 !

**Review by cisaacstern — COMMENTED** [review]
Thanks for the lighting fast ⚡ pickup on this one, @norlandrhagen!

Looking good, a few comments below re: required args.

Also, before merging could you make this change to the two kerchunk recipes in https://github.com/pangeo-forge/pangeo-forge-recipes/tree/main/examples/feedstock as well?

Thanks so much!

**Comment by norlandrhagen** [comment]
Thanks for the review @cisaacstern. Pretty sure all of the feedback is addressed!

**Comment by cisaacstern** [comment]
Thanks so much @norlandrhagen !

**Pull request merged**